.cbmc-batch/jobs/aws_cryptosdk_enc_ctx_serialize/Makefile

```diff
@@ -0,0 +1,64 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+# this file except in compliance with the License. A copy of the License is
+# located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+#########
+# if Makefile.local exists, use it. This provides a way to override the defaults 
+sinclude ../Makefile.local
+#otherwise, use the default values
+include ../Makefile.local_default
+
+MAX_TABLE_SIZE ?= 2
+
+DEFINES += -DMAX_TABLE_SIZE=$(MAX_TABLE_SIZE)
+DEFINES += -DMEMCPY_STUB_MAX=128
+
+#A table has 10 words for the struct, plus 3 words for each entry
+TABLE_SIZE_IN_WORDS=$(shell echo $$(($$((3 * $(MAX_TABLE_SIZE))) + 10)))
+
+UNWINDSET += array_list_item_generator.0:$(shell echo $$((1 + $(MAX_TABLE_SIZE))))
+UNWINDSET += aws_cryptosdk_enc_ctx_serialize.0:$(shell echo $$((1 + $(MAX_TABLE_SIZE))))
+#UNWINDSET += memcpy_using_uint64_impl.0:3
+
+CBMCFLAGS +=
+
+# Use the generator from the harness
+DEFINES += -DAWS_CRYPTOSDK_HASH_ELEMS_ARRAY_INIT_GENERATOR=array_list_item_generator
+
+ENTRY = aws_cryptosdk_enc_ctx_serialize_harness
+
+ABSTRACTIONS += $(SRCDIR)/helper-stubs/aws_array_list_sort_noop_stub.c
+ABSTRACTIONS += $(SRCDIR)/c-common-helper-stubs/aws_byte_buf_write_stub.c
+ABSTRACTIONS += $(SRCDIR)/helper-stubs/aws_cryptosdk_enc_ctx_size_stub.c
+ABSTRACTIONS += $(SRCDIR)/helper-stubs/aws_cryptosdk_hash_elems_array_init_stub.c
+
+
+ADDITIONAL_REMOVE_FUNCTION_BODY += --remove-function-body aws_cryptosdk_hash_elems_array_init
+ADDITIONAL_REMOVE_FUNCTION_BODY += --remove-function-body aws_cryptosdk_enc_ctx_size
+ADDITIONAL_REMOVE_FUNCTION_BODY += --remove-function-body aws_array_list_sort
+ADDITIONAL_REMOVE_FUNCTION_BODY += --remove-function-body __CPROVER_file_local_byte_buf_h_aws_byte_buf_write
```

Comment:
This feels like an awful lot of assumptions to make.  Can you document the reason (eg, need to remove function bodies to deal with cbmc function pointer removal, or something)?

Reply:
Done